temporal-testing/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -246,9 +252,70 @@ private StartWorkflowExecutionRequest overrideStartWorkflowExecutionRequest(
     if (taskTimeoutMillis != Durations.toMillis(request.getWorkflowTaskTimeout())) {
       request.setWorkflowTaskTimeout(Durations.fromMillis(taskTimeoutMillis));
     }
+
+    SearchAttributes.Builder searchAttributes = request.getSearchAttributes().toBuilder();
+    Map<String, Payload> payloadMap = request.getSearchAttributes().getIndexedFieldsMap();
+    for (Map.Entry<String, Payload> searchAttribute : payloadMap.entrySet()) {
+      String key = searchAttribute.getKey();
+      SearchAttributeType type = searchAttributeToType(key);
+      if (type != null && !SearchAttributeType.Unspecified.name().equals(type.name())) {
+        searchAttributes.putIndexedFields(
+            key,
+            searchAttribute
+                .getValue()
+                .toBuilder()
+                .putMetadata(
+                    EncodingKeys.METADATA_ENCODING_KEY, EncodingKeys.METADATA_ENCODING_JSON)
+                .putMetadata(EncodingKeys.METADATA_TYPE_KEY, ByteString.copyFromUtf8(type.name()))
+                .build());
+      } else {
+        searchAttributes.removeIndexedFields(key);
+        throw Status.INVALID_ARGUMENT.asRuntimeException();
+      }
+    }
+    request.setSearchAttributes(searchAttributes);
     return request.build();
   }
 
+  private SearchAttributeType searchAttributeToType(String key) {
+    SearchAttributesUtil.RegisteredSearchAttributes searchAttribute;
+    try {
+      searchAttribute = SearchAttributesUtil.RegisteredSearchAttributes.valueOf(key);
+    } catch (IllegalArgumentException e) {
+      return null;
+    }
+    switch (searchAttribute) {
+      case BatcherNamespace:
+      case BatcherUser:
+      case BinaryChecksums:
+      case ExecutionStatus:
+      case RunId:
+      case TaskQueue:
+      case TemporalChangeVersion:
+      case WorkflowId:
+      case WorkflowType:
+      case CustomKeywordField:
+      case CustomStringField:
+        return javaTypeToEncodedType(String.class);
```

Comment:
Why do we need all these calls to conversions here? Why can't we just return corresponded SearchAttributeType here? It just makes it more complex than it should be.